File: sanghyun/programmers-2/숫자 블록.java
import java.util.Arrays;

class Solution {

    public int[] solution(long begin, long end1) {
        // 범위가 어차피 int 범위 내에 있으므로 int 로 바꿔주고 계산 시작
        int start = (int) begin;
        int end = (int) end1;
        int[] answer = new int[(end - start) + 1];
        int idx = 0;

        // start end 범위 내에 answer 값을 채우는 for문
        for (int now = start; now < end + 1; now++) {
            // 소수일 경우 1
            int result = 1;
            // 2부터 Math.sqrt(now)까지 약수가 있는지 체크
            for (int i = 2; i <= Math.sqrt(now); i++) {
                // 약수일경우
                if (now % i == 0) {
                    // 최대 약수 받으려면, num / 1제외 최소 약수 = 최대 약수인데, 최대값이 천만이기 때문에
                    // 최대 약수가 천만을 넘지 않으면 최대약수를 넘김
                    if (now / i <= 10000000) {
                        result = now / i;
                        break;
                    } 
                        result = i;
                }
            }
            answer[idx++] = result;
        }
        if (start == 1) {
            answer[0] = 0;
        }
        return answer;
    }
}

File: sanghoon/programmers-2/숫자 블록.py
def solution(begin, end):
    answer = []
    for i in range(begin, end + 1):
        if i == 1 :
            answer.append(0)
        else :
            answer.append(find_value(i))
    print(answer)
def find_value(num):                        # 약수 구하는 함수
    value = 1                               # 약수가 없으면 1을 반환
    for i in range(2, int(num**(0.5))+1):   # 1의 경우는 제외한 제곱근까지 숫자 확인
        if num % i == 0:                    # 약수일 경우 value에 저장
            value = i
            if (num // i) <= 10000000:      # 천만까지만 경우의 수 따짐
                return num // i
    return value
solution(begin=1, end=10)